ts/test/session/sending/MessageQueue_test.ts

```diff
@@ -0,0 +1,202 @@
+import { expect } from 'chai';
+import * as sinon from 'sinon';
+import { GroupUtils } from '../../../session/utils';
+import { Stubs, TestUtils } from '../../../test/test-utils';
+import { MessageQueue } from '../../../session/sending/MessageQueue';
+import {
+  generateChatMessage,
+  generateClosedGroupMessage,
+  generateFakePubkey,
+  generateMemberList,
+  generateOpenGroupMessage,
+} from '../../test-utils/testUtils';
+import { OpenGroupMessage } from '../../../session/messages/outgoing';
+import { PubKey, RawMessage } from '../../../session/types';
+import { UserUtil } from '../../../util';
+import { MessageSender } from '../../../session/sending';
+import { toRawMessage } from '../../../session/utils/Messages';
+import { SessionProtocol } from '../../../session/protocols';
+import { PendingMessageCache } from '../../../session/sending/PendingMessageCache';
+
+// Equivalent to Data.StorageItem
+interface StorageItem {
+  id: string;
+  value: any;
+}
+
+describe('MessageQueue', () => {
+  // Initialize new stubbed cache
+  let data: StorageItem;
+  const sandbox = sinon.createSandbox();
+  const ourNumber = generateFakePubkey().key;
+
+  // Keep track of Session Requests in each test
+  let sessionRequestSent: boolean;
+
+  // Initialize new stubbed cache
+  let messageQueueStub: MessageQueue;
+  // Message Sender Stubs
+  let sendStub: sinon.SinonStub<[RawMessage, (number | undefined)?]>;
+  let sendToOpenGroupStub: sinon.SinonStub<[OpenGroupMessage]>;
+  // Group Utils Stubs
+  let isMediumGroupStub: sinon.SinonStub;
+  let groupMembersStub: sinon.SinonStub;
+  // Session Protocol Stubs
+  let hasSessionStub: sinon.SinonStub;
+
+  beforeEach(async () => {
+    // Stub out methods which touch the database
+    const storageID = 'pendingMessages';
+    data = {
+      id: storageID,
+      value: '[]',
+    };
+
+    // Pending Message Cache Data Stubs
+    TestUtils.stubData('getItemById')
```

Comment:
Instead of doing this we can just stub out the pending message cache.
```
sandbox.createStubbedInstance(PendingMessageCache)
```
This way we don't really have to worry about the implementation of the cache in this test